test/test_autograd.py

```diff
@@ -3640,6 +3640,22 @@ def fn(sparse):
         check(fast_mode=True)
         check(fast_mode=False)
 
+    @unittest.expectedFailure
+    def test_gradcheck_sparse_csr_input(self):
```

Comment:
Note that to reduce test file growth, we started a `test/autograd` subfolder: https://github.com/pytorch/pytorch/tree/master/test/autograd

Feel free to add a sparse_csr file there to do all the autograd related testing.
Note that everything in that folder should still run when `test/test_autograd.py` is run. This split is only for readability.

Reply:
Thanks! It's good to know, I will make use of that folder in the future. I put this test here so that it's near `test_gradcheck_sparse_input` (which is located above this test).

Reply:
Yes for just that test I think this is ok.
We might want to do a cleanup and move all of the sparse related stuff out though in the future :)